### ISSUE TITLE
Intel Fortran compiler detection fails on Mac OS X 10.6 (migrated from Trac #1448)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3000]
Original ticket http://projects.scipy.org/numpy/ticket/1448
Reported 2010-04-08 by trac user themiwi, assigned to unknown.

Running the version_cmd (as in numpy/distutils/ccompiler.py) results in
    $ /usr/bin/ifort -FI -V -c test.f -o test.o 
    Intel(R) Fortran Intel(R) 64 Compiler Professional for applications running on Intel(R) 64, Version 11.1    Build 20090827 Package ID: m_cprof_p_11.1.067
    Copyright (C) 1985-2009 Intel Corporation.  All rights reserved.

```
 Intel(R) Fortran 11.1-2582
```

Using the following patch fixes things for me, but probably will break other platforms, so somebody with a better knowledge of the whole compiler-detection code should take care of this.

```
Index: numpy/distutils/fcompiler/intel.py
===================================================================
--- numpy/distutils/fcompiler/intel.py  (revision 8324)
+++ numpy/distutils/fcompiler/intel.py  (working copy)
atmention:atmention: -24,8 +24,8 atmention:atmention:

     compiler_type = 'intel'
     compiler_aliases = ('ifort',)
-    description = 'Intel Fortran Compiler for 32-bit apps'
-    version_match = intel_version_match('32-bit|IA-32')
+    description = 'Intel Fortran Compiler for 32/64-bit apps'
+    version_match = intel_version_match('(32-bit|64)|IA-(32|64)')

     possible_executables = ['ifort', 'ifc']
```

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-07-18

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-07-18

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-11-30

Do you have a problem using the 'intelem' compiler? It matches your current version string fine and should just work. If not, we can consider adding the below {{{v64match}}}}. It's shouldn't break anything, while your version could break on 32-bit.

I do think the name 'intelem', especially the description 'Intel Fortran Compiler for EM64T-based apps' is quite old-fashioned. I had to look up EM64T to see it's just plain 64 bit.
    In [66]: str32 = "Intel(R) Fortran Intel(R) 32-bit Compiler Professional for applications running on Intel(R) 32, Version 11.1"

```
In [67]: str64 = "Intel(R) Fortran Intel(R) 64 Compiler Professional for applications running on Intel(R) 64, Version 11.1"

In [68]: v64match = intel_version_match('32-bit|IA-32|64|IA-64')

In [69]: v32match = intel_version_match('32-bit|IA-32')

In [70]: emt64_version_match = intel_version_match('EM64T-based|Intel\\(R\\) 64')

In [71]: v32match(None, str32)  # current standard 32-bit match
Out[71]: '11.1'

In [72]: v64match(None, str64)  # proposed new match
Out[72]: '11.1'

In [73]: v64match(None, str32)  # proposed new match works for 32-bit
Out[73]: '11.1'

In [74]: emt64_version_match(None, str64)  # this works already
Out[74]: '11.1'
```

Status is needs_info, but Trac won't let me set that.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-12-12

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-03-03

From #1145 (closing as duplicate of this one):
    Intel(R) Fortran IA-64 Compiler Professional for applications running on IA-64, Version 11.0    Build 20081105 Package ID: l_cprof_p_11.0.074
    Copyright (C) 1985-2008 Intel Corporation.  All rights reserved.

```
Intel Fortran 11.0-1558
```

**Comment by thouis** [comment]
Comment in Trac by trac user themiwi, 2011-03-03

Replying to [comment:3 rgommers]:

> Do you have a problem using the 'intelem' compiler? It matches your current version string fine and should just work. If not, we can consider adding the below {{{v64match}}}}. It's shouldn't break anything, while your version could break on 32-bit.
> 
> I do think the name 'intelem', especially the description 'Intel Fortran Compiler for EM64T-based apps' is quite old-fashioned. I had to look up EM64T to see it's just plain 64 bit.

Sorry for the long delay. I can't check since I switched back to Linux... But using intelem would be fine, as long as the description would be updated. EM64T is really outmoded perhaps could be replaced by more commonly used terms, such as "x86_64", "amd64" or "Intel 64". I'd prefer the first, since that's widely used and vendor neutral. The second is used by Linux and GCC, but I find it to be misleading. The same goes for "Intel 64".

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-03-09

Fixed in commit:ce9fea2
